pkg/virt-controller/services/template.go

```diff
@@ -359,17 +361,14 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 
 	ovmfPath := t.clusterConfig.GetOVMFPath(vmi.Spec.Architecture)
 
-	// Read requested hookSidecars from VMI meta
-	requestedHookSidecarList, err := hooks.UnmarshalHookSidecarList(vmi)
-	if err != nil {
-		return nil, err
-	}
-
-	bindingSidecars, err := NetBindingPluginSidecarList(vmi, t.clusterConfig.GetConfig(), t.recorder)
-	if err != nil {
-		return nil, err
+	var requestedHookSidecarList hooks.HookSidecarList
+	for _, sidecarCreator := range t.sidecarCreators {
+		sidecars, serr := sidecarCreator(vmi, t.clusterConfig.GetConfig())
```

Comment:
When serr occur err is returned.
Consider getting rid of `serr` and use `err` instead.

Reply:
I did it to avoid shadowing the original `err`, as some linters do not like it.
But I guess the bug you detected here shows it is not always good to do it, as it opens up the opportunity for other problems/bugs.

Done